xapian-letor/api/letor_internal.cc

```diff
@@ -123,16 +113,18 @@ Letor::Internal::load_list_fvecs(const char *filename) { //directly use train.tx
         train_file >> qid;
 
         qid = qid.substr(qid.find(":")+1,qid.length());
+        train_file.ignore();
 
-        std::map<int,double> fvals;//read features
-        for(int i = 1; i < 20; ++i){
-            train_file.ignore();
+        std::vector<double> fvals;
+
+        while (train_file.peek() != 35) { // read till '#docid' is found (35 is ascii for '#')
             int feature_index;
             double feature_value;
             train_file >> feature_index;
             train_file.ignore();
             train_file >> feature_value;
-            fvals[feature_index] = feature_value;
+            fvals.push_back(feature_value);
+            train_file.ignore();
```

Comment:
:-)
On 12 Aug 2016 9:00 p.m., Ayush Tomar <notifications@github.com> wrote:In xapian-letor/api/letor_internal.cc:
>              int feature_index;
>              double feature_value;
>              train_file >> feature_index;
>              train_file.ignore();
>              train_file >> feature_value;
> -            fvals[feature_index] = feature_value;
> +            fvals.push_back(feature_value);
> +            train_file.ignore();

Its probably verbose to excess now

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.